integration/layer_reuse_test.go

```diff
@@ -0,0 +1,231 @@
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it.After(func() {
+			for containerID := range containerIDs {
+				Expect(docker.Container.Remove.Execute(containerID)).To(Succeed())
+			}
+
+			for imageID := range imageIDs {
+				Expect(docker.Image.Remove.Execute(imageID)).To(Succeed())
+			}
+
+			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it("reuses the cached runtime layer", func() {
+			var err error
+			source, err = occam.Source(filepath.Join("testdata", "default"))
+			Expect(err).NotTo(HaveOccurred())
+
+			var logs fmt.Stringer
+			firstImage, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					settings.Buildpacks.DotnetCoreAspnetRuntime.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			imageIDs[firstImage.ID] = struct{}{}
+
+			Expect(firstImage.Buildpacks).To(HaveLen(2))
+			Expect(firstImage.Buildpacks[0].Key).To(Equal(buildpackInfo.Buildpack.ID))
+			Expect(firstImage.Buildpacks[0].Layers).To(HaveKey("dotnet-core-aspnet-runtime"))
+
+			// second pack build
+
+			secondImage, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					settings.Buildpacks.DotnetCoreAspnetRuntime.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			imageIDs[secondImage.ID] = struct{}{}
+
+			Expect(secondImage.Buildpacks).To(HaveLen(2))
+			Expect(secondImage.Buildpacks[0].Key).To(Equal(buildpackInfo.Buildpack.ID))
+			Expect(secondImage.Buildpacks[0].Layers).To(HaveKey("dotnet-core-aspnet-runtime"))
+
+			Expect(logs).To(ContainLines(
+				"  Resolving .NET Core ASP.NET Runtime version",
+				"    Candidate version sources (in priority order):",
+				`      runtimeconfig.json -> "6.0.0"`,
+				`      <unknown>          -> ""`,
+				"",
+				"    No exact version match found; attempting version roll-forward",
+				"",
+				MatchRegexp(`    Selected .NET Core ASP.NET Runtime version \(using runtimeconfig.json\): 6\.0\.\d+`),
+				"",
+				MatchRegexp(fmt.Sprintf("  Reusing cached layer /layers/%s/dotnet-core-aspnet-runtime", strings.ReplaceAll(buildpackInfo.Buildpack.ID, "/", "_"))),
+				"",
+			))
+
+			secondContainer, err = docker.Container.Run.
+				WithCommand("dotnet --info").
+				Execute(secondImage.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			containerIDs[secondContainer.ID] = struct{}{}
+
+			Eventually(func() string {
+				cLogs, err := docker.Container.Logs.Execute(secondContainer.ID)
+				Expect(err).NotTo(HaveOccurred())
+				return cLogs.String()
+			}).Should(ContainSubstring(".NET runtimes installed"))
+
+			Expect(secondImage.Buildpacks[0].Layers["dotnet-core-aspnet-runtime"].SHA).To(Equal(firstImage.Buildpacks[0].Layers["dotnet-core-aspnet-runtime"].SHA))
+		})
+	})
```

Comment:
I think this is an opportunity to use the new `BuildpackForKey` helper
https://github.com/paketo-buildpacks/occam/pull/165